### ISSUE TITLE
Improve GoDoc of top level package

### ISSUE THREAD

**Issue by philippgille** [cockroachdb/errors#6]
I wanted to see an overview of the available functions and their descriptions in a format I'm used to -  via GoDocs. And thought I could even send a small PR that adds the GoDoc badge to the README, like this: [![GoDoc](https://godoc.org/github.com/cockroachdb/errors?status.svg)](https://godoc.org/github.com/cockroachdb/errors).

But the GoDoc of the top level package, which is the one most people are going to use, doesn't contain much useful info. Almost all of the functions just say "... forwards a definition". See https://godoc.org/github.com/cockroachdb/errors.

I think it would be useful if the functions' GoDoc contain the proper information. For example the GoDoc of `errors.WithStack(error)` (https://github.com/cockroachdb/errors/blob/9b2e535fbedbf187adf0de1d98509ae8a70ae44e/withstack_api.go#L19-L20) only says "WithStack forwards a definition.", but the README contains much more useful info here, like:

> usually not needed, use errors.Wrap()/errors.Wrapf() instead.

On the other hand, in the subpackages, the functions are much better documented. Sticking with the previous example of `WithStack`: https://github.com/cockroachdb/errors/blob/78b6caa8a0455b5fa93f17d4b4c3132597342c0e/withstack/withstack.go#L23-L36

---

TL;DR: I'd love to see the top level package documented in the same way as the subpackages.

**Comment by knz** [comment]
And how do you propose to do this?

**Comment by philippgille** [comment]
Sorry I don't understand your question. I created the issue quite a while ago so I don't remember the details, but as a first time CockroachDB user I was checking out the GoDoc and was put off by the strange "forwards a definition" comments everywhere. Why is there no proper GoDoc for each public function? Are the comments auto-generated or something?

**Comment by knz** [comment]
Oh my apologies I hadn't realized that the issue was filed a few month ago.

So the issue here is that the locus of work for engineers is the sub-package. That's where the logic is present and thus where it's most natural to produce the explanatory comments.

Meanwhile the existence of top-level API forwards is for convenience for users of the library.

I understand the mismatch but I am not sure how to resolve it. Is there a way to teach godoc to take over a comment from a function defined elsewhere?

**Comment by philippgille** [comment]
I don't know of such a mechanism (taking over comment from another function). Would it be too bad to just copy the comments manually and then when modifying a comment in a subpackage the engineer should always make sure to also modify the top-level package's comment?

Or maybe a different content for the comments actually makes sense. This depends though. But you say the subpackage is for the engineers, so the comments might say more about the inner workings of a function. While the top-level package is for libary users. Those functions could contain more info about how to *use* the functions.
